File: Backend/Iteramico/DomainRepository/Models/Event.cs
namespace DomainRepository.Models
{
#nullable disable
    public class Event
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime From { get; set; }
        public DateTime To { get; set; }
        public virtual User Creator { get; set; }
        public virtual Journey Journey { get; set; }
    }
}

File: frontend/lib/models/event.dart
import 'package:frontend/models/user.dart';

class Event {
  final int id;
  final String name;
  final DateTime from;
  final DateTime to;
  final User creator;

  Event(
      {required this.id,
      required this.name,
      required this.from,
      required this.to,
      required this.creator});

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
        id: json['id'],
        name: json['name'],
        from: DateTime.parse(json['from']),
        to: DateTime.parse(json['to']),
        creator: User.fromJson(json['creator']));
  }
}
